pkg/install/vault/policy.go

```diff
@@ -0,0 +1,152 @@
+package vault
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/hashicorp/hcl2/gohcl"
+	"github.com/hashicorp/hcl2/hclwrite"
+)
+
+// vaultPolicy is a subset of the vault.Policy model. We created this here for
+// 2 reasons: vault is a big module that we want to prevent pulling in as much
+// as possible (the sdk and api are usually okay) and the model isn't made for
+// encoding, so it doesn't include the proper hcl tags.
+type vaultPolicy struct {
+	Paths []vaultPolicyPath `hcl:"path,block"`
+}
+
+type vaultPolicyPath struct {
+	Name         string   `hcl:"name,label"`
+	Capabilities []string `hcl:"capabilities"`
+}
+
+type vaultPolicyGenerator struct {
+	AuthJWTAccessor string
+	LogServicePath  string
+	TenantPath      string
+	TransitPath     string
+	TransitKey      string
+}
+
+func (g *vaultPolicyGenerator) operatorPolicy() []byte {
+	policy := vaultPolicy{
+		Paths: []vaultPolicyPath{
+			{
+				Name:         path.Join(g.TransitPath, "encrypt", g.TransitKey),
+				Capabilities: []string{"update"},
+			},
+			{
+				Name:         path.Join("sys", "mounts"),
+				Capabilities: []string{"read"},
+			},
+			{
+				Name:         path.Join(g.TenantPath, "data", "workflows", "*"),
+				Capabilities: []string{"create", "update"},
+			},
+			{
+				Name:         path.Join(g.TenantPath, "data", "connections", "*"),
+				Capabilities: []string{"create", "update"},
+			},
+			{
+				Name:         path.Join("oauth", "+", "creds", "*"),
+				Capabilities: []string{"read", "create", "update", "delete"},
+			},
+			{
+				Name:         path.Join("oauth", "auth0-management", "self", "management-api"),
+				Capabilities: []string{"read"},
+			},
+			{
+				Name:         path.Join(g.TenantPath, "metadata", "workflows", "*"),
+				Capabilities: []string{"list", "delete"},
+			},
+			{
+				Name:         path.Join(g.TenantPath, "metadata", "connections", "*"),
+				Capabilities: []string{"list", "delete"},
+			},
+			{
+				Name:         path.Join("customers-oauth", "auth-code-url"),
+				Capabilities: []string{"update"},
+			},
+			{
+				Name:         path.Join("customers-oauth/creds/*"),
+				Capabilities: []string{"read", "create", "update", "delete"},
+			},
```

Comment:
I believe these policies are also all specific to the API and the operator shouldn't need any of them. The only policy the operator itself should need is encrypting through the transit path.

Reply:
Ahh, yeah, likely another circumstance where the `relay` dev environment was setting these up in advance (these should have been in the API-specific workflow then). I'll remove for now and we'll see about moving this overall logic into an API-specific image for the workflow.